node-video-crime-miner/src/AWS Layer/Rekognition/videoLabelUtils.ts

```diff
@@ -17,20 +18,24 @@ var attributes = {
 }
 const client  = new RekognitionClient(attributes)
 
-async function startLabelDetection(bucketName:string, videoName:string) {
+async function startLabelDetection(bucketName:string, videoName:string, snsTopicArn:string, clientToUse:RekognitionClient | any=client) {
   try {
     var attributes = {
       Video: { 
         S3Object: { 
           Name: videoName,
           Bucket: bucketName,
         }
+      },
+      NotificationChannel:{
+        RoleArn: roleArn, 
+        SNSTopicArn: snsTopicArn
       }
     }
 
     // Returns jobId to get when it's finished by getVideoFacesDetectionOutput
     const command = new StartLabelDetectionCommand(attributes)
-    const result = await client.send(command)
+    const result = await clientToUse.send(command)
```

Comment:
Why is clientToUse a parameter? Can we not just use the client variable?

Reply:
Each file has a legit aws service client (like 'const client = new SNSClient(attributes)' in snsClient.ts or 'const client  = new RekognitionClient(attributes)' in videoLabelUtils)

For the purposes of testing, I changed the methods so that you can pass a mocked client instead, like in each of the tests of the .test.ts files, you can see that each methods is passing the mocked version.

Now that I'm taking another look at the face data stuff, I think that it is not mocked right. It looks like the client is mocked for it in the tests as 'rekognitionMock', but then the line 'const result = await startVideoFacesDetection(inputBucket, inputFile)' is just going to use 'const client  = new RekognitionClient(attributes)' from videoFaceUtils.ts